### PR TITLE
ui: add community submission request template

### PR DESCRIPTION
**Pull request by kpsherva**

### CODE REVIEW

File: invenio_app_rdm/theme/templates/semantic-ui/invenio_requests/community-submission/index.html [hunk 1]
```diff
@@ -0,0 +1,58 @@
+{# -*- coding: utf-8 -*-
+
+  This file is part of Invenio.
+  Copyright (C) 2016-2020 CERN.
```
Comment (jrcastro2) [inline invenio_app_rdm/theme/templates/semantic-ui/invenio_requests/community-submission/index.html:4]: 2022

File: invenio_app_rdm/theme/templates/semantic-ui/invenio_requests/community-submission/index.html [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+  <h1 class="ui header">
+    <div class="ui label green">New submission</div>
+    {{ request.topic.metadata.title }}</h1><br/>
+  <div class="ui horizontal list">
+    {% for item in request.topic.metadata.creators %}
+      <div class="item">
+        <div class="content"><i class="user icon"></i>
+          {{ item.person_or_org.name }}</div>
+      </div>
+    {% endfor %}
```
Comment (jrcastro2) [inline invenio_app_rdm/theme/templates/semantic-ui/invenio_requests/community-submission/index.html:29]: Is it possible to have many authors? 
If yes, maybe we should add an expand button or limit the displayed creators?

Comment (ntarocco) [reply invenio_app_rdm/theme/templates/semantic-ui/invenio_requests/community-submission/index.html:29]: See similar issue: https://github.com/inveniosoftware/invenio-communities/issues/401

Comment (kpsherva) [reply invenio_app_rdm/theme/templates/semantic-ui/invenio_requests/community-submission/index.html:29]: I will create an issue for this, because the PR is growing too big
btw, I think it should be initially fixed in the record_ui templates. I will suggest that in the ticket

File: invenio_app_rdm/theme/templates/semantic-ui/invenio_requests/community-submission/index.html [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+  </div>
+
+  <div class="ui bottom attached tab segment active" data-tab="conversation">
+    <div class="ui divider hidden"></div>
+    {{ super() }}
+  </div>
+  <div class="ui bottom attached tab segment" data-tab="record">
+  <div class="ui divider hidden"></div>
+    {% if record %}
+      {% set standalone = true %}
```
[the remark this answers is not in the document]
Comment (kpsherva) [reply invenio_app_rdm/theme/templates/semantic-ui/invenio_requests/community-submission/index.html:52]: I think we cannot, otherwise the base template header and footer will be visible in the tab.
Unless there is another way to do it, without maintaining two templates, but I didn't find any. Even with an iframe we would have to implement something similar, I found this one the safest. And the backwards compatibility is maintained for this flag

[the remark this answers is not in the document]
Comment (kpsherva) [reply invenio_app_rdm/theme/templates/semantic-ui/invenio_requests/community-submission/index.html]: I renamed the flag as discussed

File: invenio_app_rdm/theme/assets/semantic-ui/js/invenio_app_rdm_requests/menuTabInit.js [hunk 4]
```diff
@@ -0,0 +1,3 @@
+import $ from "jquery";
+
+$('#request-community-submission-tab').tab();
```
Comment (jennur) [inline invenio_app_rdm/theme/assets/semantic-ui/js/invenio_app_rdm_requests/menuTabInit.js:3]: Make sure tab menus are accessible, see example: https://github.com/inveniosoftware/invenio-app-rdm/blob/master/invenio_app_rdm/theme/assets/semantic-ui/js/invenio_app_rdm/landing_page/index.js#L112

File: invenio_app_rdm/theme/templates/semantic-ui/invenio_requests/community-submission/index.html [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+    {% endfor %}
+  </div>
+{% endblock request_topic %}
+
+{% block conversation %}
+  <div class="ui divider hidden"></div>
+  <div class="ui container">
+  <div class="ui top attached tabular menu" id="request-community-submission-tab">
+    <a class="active item" data-tab="conversation">
+      Conversation
```
Comment (jennur) [inline invenio_app_rdm/theme/templates/semantic-ui/invenio_requests/community-submission/index.html:36]: Please add aria-attributes. Example for tabs: 
https://github.com/inveniosoftware/invenio-app-rdm/blob/master/invenio_app_rdm/records_ui/templates/semantic-ui/invenio_app_rdm/records/details/details.html#L22

File: invenio_app_rdm/theme/templates/semantic-ui/invenio_requests/community-submission/index.html [hunk 6]
```diff
@@ -0,0 +1,56 @@
...
+    </a>
+    <a class="item" data-tab="record">
+      Record
+    </a>
+  </div>
+
+  <div class="ui bottom attached tab segment active" data-tab="conversation">
```
Comment (jennur) [inline invenio_app_rdm/theme/templates/semantic-ui/invenio_requests/community-submission/index.html:43]: Aria attributes for tab panel example: 
https://github.com/inveniosoftware/invenio-app-rdm/blob/master/invenio_app_rdm/records_ui/templates/semantic-ui/invenio_app_rdm/records/details/details.html#L110

**Review by jennur — COMMENTED** [review]
Just added a few comments on some accessibility improvements! :)

File: invenio_app_rdm/theme/assets/semantic-ui/js/invenio_app_rdm/user_dashboard/components/requests.js [hunk 7]
```diff
@@ -128,7 +128,7 @@ export const RequestsResults = ({
 
 export function RequestsResultsGridItemTemplate({ result, index }) {
   return (
-    <Card fluid key={index} href={`/requests/${result.metadata.id}`}>
+    <Card fluid key={index} href={`/me/requests/${result.metadata.id}`}>
```
Comment (Pineirin) [inline invenio_app_rdm/theme/assets/semantic-ui/js/invenio_app_rdm/user_dashboard/components/requests.js:131]: Shouldn't this be obtained through the user-dashboard-community-submission variable defined in the config.py

File: invenio_app_rdm/requests_ui/views/requests.py [hunk 8]
```diff
@@ -0,0 +1,82 @@
...
+
+    if 'record' in request_dict["topic"]:
+        try:
+            request_dict["topic"] = request._request.topic.resolve()
+            draft = request_dict["topic"]
+
+            topic_record = current_rdm_records_service.read_draft(
+                id_=draft["id"], identity=g.identity)
+
+        except NoResultFound:
```
Comment (Pineirin) [inline invenio_app_rdm/requests_ui/views/requests.py:52]: Will we delete this exceptions once the actions issue is solved?

**Comment by ntarocco** [comment]
@jennur thank you for your feedback! It is clear that we are missing some cross-team communication, we need to address that.

File: invenio_app_rdm/config.py [hunk 9]
```diff
@@ -661,3 +661,7 @@ def files_rest_permission_factory(obj, action):
     'sort': ['bestmatch', 'newest', 'oldest'],
 }
 """User requests search configuration (i.e list of user requests)"""
+
+RDM_REQUESTS_ROUTES = {
+    'user-dashboard-community-submission': '/me/requests/<pid_value>',
```
Comment (ntarocco) [inline invenio_app_rdm/config.py:666]: I am not sure this is about the dashboard (search). Isn't it about the request details page?
```suggestion
    'user-requests-community-submission': '/me/requests/<pid_value>',
```

File: invenio_app_rdm/records_ui/templates/semantic-ui/invenio_app_rdm/records/detail.html [hunk 10]
```diff
@@ -9,7 +9,9 @@
   it under the terms of the MIT License; see LICENSE file for more details.
 #}
 
+{% if not drop_theme_basic_template %}
```
Comment (ntarocco) [inline invenio_app_rdm/records_ui/templates/semantic-ui/invenio_app_rdm/records/detail.html:12]: It looks to me like a double negation `not` + `drop`. Can we convert it to positve instead?

File: invenio_app_rdm/requests_ui/views/requests.py [hunk 11]
```diff
@@ -0,0 +1,82 @@
...
+from sqlalchemy.orm.exc import NoResultFound
+from flask import g
+from invenio_requests.views.decorators import pass_request
+from invenio_rdm_records.resources.serializers import UIJSONSerializer
+from invenio_rdm_records.proxies import current_rdm_records_service
+
+
+@login_required
+@pass_request
+def requests_detail(request=None, pid_value=None):
```
Comment (ntarocco) [inline invenio_app_rdm/requests_ui/views/requests.py:23]: what happens if I try to see the request of another user? Is the `pass_request` checking that?

File: invenio_app_rdm/requests_ui/views/requests.py [hunk 12]
```diff
@@ -0,0 +1,82 @@
...
+        request_dict["created_by"].update({
+            "avatar": "/static/images/placeholder.png",
+            "full_name": request_dict["created_by"]["community"]
+        })
+
+    topic_record = None
+    is_draft = None
+    permissions = []
+
+    if 'record' in request_dict["topic"]:
```
Comment (ntarocco) [inline invenio_app_rdm/requests_ui/views/requests.py:44]: Can you change this to a named constant? what does it mean "having a record in a topic"?

File: invenio_app_rdm/requests_ui/views/requests.py [hunk 13]
```diff
@@ -0,0 +1,82 @@
...
+        topic_record = UIJSONSerializer().serialize_object_to_dict(
+            topic_record.data)
+        request_dict["topic"] = topic_record
+
+    # end temporary block
+
+    try:
+        return render_template(
+            f"invenio_requests/{request_dict['type']}/index.html",
+            request=request_dict,  # TODO: use serializer
```
Comment (ntarocco) [inline invenio_app_rdm/requests_ui/views/requests.py:69]: the TODO does not really explain how the code should be changed. Can you please elaborate?

File: invenio_app_rdm/requests_ui/views/ui.py [hunk 14]
```diff
@@ -0,0 +1,43 @@
...
+from invenio_records_resources.services.errors import PermissionDeniedError
+
+from invenio_app_rdm.requests_ui.views.requests import requests_detail
+from invenio_requests.views.ui import record_permission_denied_error, \
+    record_tombstone_error, not_found_error
+
+
+def create_ui_blueprint(app):
+    """Register blueprint routes on app."""
+    routes = app.config.get("RDM_REQUESTS_ROUTES")
```
Comment (ntarocco) [inline invenio_app_rdm/requests_ui/views/ui.py:22]: Why the `.get`?
```suggestion
    routes = app.config["RDM_REQUESTS_ROUTES"]
```

File: invenio_app_rdm/theme/templates/semantic-ui/invenio_requests/community-submission/index.html [hunk 15]
```diff
@@ -0,0 +1,56 @@
...
+    <div class="ui divider hidden"></div>
+    {{ super() }}
+  </div>
+  <div class="ui bottom attached tab segment" data-tab="record">
+  <div class="ui divider hidden"></div>
+    {% if record %}
+      {% set drop_theme_basic_template = true %}
```
Comment (ntarocco) [inline invenio_app_rdm/theme/templates/semantic-ui/invenio_requests/community-submission/index.html:50]: Naming not clear to me, sorry :(

File: invenio_app_rdm/theme/assets/semantic-ui/js/invenio_app_rdm/user_dashboard/components/requests.js (hunk 7 above)
Comment (kpsherva) [reply invenio_app_rdm/theme/assets/semantic-ui/js/invenio_app_rdm/user_dashboard/components/requests.js:131]: there is a separate issue created for api urls creation, which is not the scope of this PR

File: invenio_app_rdm/config.py (hunk 9 above)
Comment (kpsherva) [reply invenio_app_rdm/config.py:666]: as far as I understood, there will be different endpoints for different ways of reaching the request details page, this is only one of them (user dashboard)
another one is for example community requests (which will end up as something similar to `/<community>/requests/<pid>`
Ping @lnielsen

File: invenio_app_rdm/requests_ui/views/requests.py (hunk 11 above)
Comment (kpsherva) [reply invenio_app_rdm/requests_ui/views/requests.py:23]: the service is checking it, and it will return 403

File: invenio_app_rdm/theme/assets/semantic-ui/js/invenio_app_rdm_requests/menuTabInit.js (hunk 4 above)
Comment (kpsherva) [reply invenio_app_rdm/theme/assets/semantic-ui/js/invenio_app_rdm_requests/menuTabInit.js:3]: the rule from the example you have linked is applied already, because it is connected with the class selector 
`$(.menu.tab)`
however, because there might be further changes, I will add this event to my element. 
In general I believe this part should be refactored, and placed in a common module, since it should be applied globally per theme

File: invenio_app_rdm/requests_ui/views/requests.py (hunk 8 above)
Comment (kpsherva) [reply invenio_app_rdm/requests_ui/views/requests.py:52]: not exactly, it will be handled differently, inside a serializer.

File: invenio_app_rdm/config.py (hunk 9 above)
Comment (lnielsen) [reply invenio_app_rdm/config.py:666]: Correct

File: invenio_app_rdm/records_ui/templates/semantic-ui/invenio_app_rdm/records/detail.html [hunk 16]
```diff
@@ -9,7 +9,9 @@
   it under the terms of the MIT License; see LICENSE file for more details.
 #}
 
+{% if use_theme_basic_template|default(true) %}
 {%- extends config.BASE_TEMPLATE %}
+{% endif %}
```
Comment (Pineirin) [inline invenio_app_rdm/records_ui/templates/semantic-ui/invenio_app_rdm/records/detail.html:14]: Maybe missing indent?

File: invenio_app_rdm/records_ui/templates/semantic-ui/invenio_app_rdm/records/detail.html [hunk 17]
```diff
@@ -273,7 +275,11 @@ <h2 id="files-heading">{{ _('Files') }}</h2>
 {%- endblock page_body %}
 
 {%- block javascript %}
+{% if use_theme_basic_template|default(true) %}
 {{ super() }}
+{% else %}
+{% include config.THEME_JAVASCRIPT_TEMPLATE %}
+{% endif %}
```
Comment (Pineirin) [inline invenio_app_rdm/records_ui/templates/semantic-ui/invenio_app_rdm/records/detail.html:282]: Same than above?

File: invenio_app_rdm/theme/assets/semantic-ui/js/invenio_app_rdm_requests/menuTabInit.js [hunk 18]
```diff
@@ -0,0 +1,22 @@
+import $ from "jquery";
+
+const tabElementSelector = "#request-community-submission-tab .item";
+const tabElement = $(tabElementSelector);
```
Comment (ntarocco) [inline invenio_app_rdm/theme/assets/semantic-ui/js/invenio_app_rdm_requests/menuTabInit.js:4]: Super minor, no change needed, for the future:
```suggestion
const $tabElement = $(tabElementSelector);
```
so that you can immediately see that it is a `jQuery` obj

File: invenio_app_rdm/theme/assets/semantic-ui/js/invenio_app_rdm_requests/menuTabInit.js [hunk 19]
```diff
@@ -0,0 +1,22 @@
...
+
+tabElement.tab({
+  onVisible: function (tab) {
+    $(tabElementSelector).attr("aria-selected", false);
+    $(`#${tab}-tab-menu`).attr("aria-selected", true);
+
+    $(` .tab.segment`).attr("hidden", true);
```
Comment (ntarocco) [inline invenio_app_rdm/theme/assets/semantic-ui/js/invenio_app_rdm_requests/menuTabInit.js:11]:
```suggestion
    $('.tab.segment').attr("hidden", true);
```
it this line correct? It will look for all the `.tab.segment` elements.

Suggestion: to make it more efficient, it would be nice to wrap all `.tab.segment` inside an `id` so that we can do:
```
$('#myid').find('.tab.segment').attr("hidden", true);
```

File: invenio_app_rdm/theme/assets/semantic-ui/js/invenio_app_rdm_requests/menuTabInit.js [hunk 20]
```diff
@@ -0,0 +1,22 @@
...
+    $(`#${tab}-tab`).attr("hidden", false);
+  },
+});
+
+tabElement.on("keydown", function (event) {
```
Comment (ntarocco) [inline invenio_app_rdm/theme/assets/semantic-ui/js/invenio_app_rdm_requests/menuTabInit.js:16]: Why is this needed? Can you add a comment?

**Review by ntarocco — APPROVED** [review]
